SDWebImage/SDImageCache.m

```diff
@@ -217,50 +217,70 @@ - (void)storeImage:(nullable UIImage *)image
     
     if (toDisk) {
         dispatch_async(self.ioQueue, ^{
+            NSError * writeError = nil;
             @autoreleasepool {
                 NSData *data = imageData;
                 if (!data && image) {
                     SDImageFormat imageFormatFromData = [NSData sd_imageFormatForImageData:data];
                     data = [image sd_imageDataAsFormat:imageFormatFromData];
                 }                
-                [self storeImageDataToDisk:data forKey:key];
+                [self storeImageDataToDisk:data forKey:key error:&writeError];
             }
             
             if (completionBlock) {
                 dispatch_async(dispatch_get_main_queue(), ^{
-                    completionBlock();
+                    completionBlock(writeError);
                 });
             }
         });
     } else {
         if (completionBlock) {
-            completionBlock();
+            completionBlock(nil);
         }
     }
 }
 
-- (void)storeImageDataToDisk:(nullable NSData *)imageData forKey:(nullable NSString *)key {
+- (void)storeImageDataToDisk:(nullable NSData *)imageData
+                      forKey:(nullable NSString *)key {
+    [self storeImageDataToDisk:imageData forKey:key fileManager:_fileManager error:nil];
+}
+
+- (BOOL)storeImageDataToDisk:(nullable NSData *)imageData
+                      forKey:(nullable NSString *)key
+                       error:(NSError * _Nullable * _Nullable)errorPtr {
+    return [self storeImageDataToDisk:imageData forKey:key fileManager:_fileManager error:errorPtr];
+}
+
+- (BOOL)storeImageDataToDisk:(nullable NSData *)imageData
+                      forKey:(nullable NSString *)key
+                 fileManager:(nullable NSFileManager *)fileManager
+                       error:(NSError * _Nullable * _Nullable)errorPtr {
     if (!imageData || !key) {
-        return;
+        return YES;
     }
     
     [self checkIfQueueIsIOQueue];
     
-    if (![_fileManager fileExistsAtPath:_diskCachePath]) {
-        [_fileManager createDirectoryAtPath:_diskCachePath withIntermediateDirectories:YES attributes:nil error:NULL];
+    if (![fileManager fileExistsAtPath:_diskCachePath]) {
+        [fileManager createDirectoryAtPath:_diskCachePath withIntermediateDirectories:YES attributes:nil error:NULL];
     }
     
     // get cache Path for image key
     NSString *cachePathForKey = [self defaultCachePathForKey:key];
     // transform to NSUrl
     NSURL *fileURL = [NSURL fileURLWithPath:cachePathForKey];
     
-    [_fileManager createFileAtPath:cachePathForKey contents:imageData attributes:nil];
+    if (![fileManager createFileAtPath:cachePathForKey contents:imageData attributes:nil] && errorPtr) {
```

Comment:
Apple says `/* These methods are provided here for compatibility. The corresponding methods on NSData which return NSErrors should be regarded as the primary method of creating a file from an NSData or retrieving the contents of a file as an NSData.
 */`
So, I suggest to use NSData's `writeToFile:options:error:` to write to file, but not just grap the error code from POSIX file API and give back to user.